ggml/src/ggml.c

```diff
@@ -320,6 +329,11 @@ static ggml_fp16_t ggml_table_gelu_quick_f16[1 << 16];
 // precomputed f32 table for f16 (256 KB) (ggml-impl.h)
 float ggml_table_f32_f16[1 << 16];
 
+#if GGML_USE_AMX
+// global flag for amx init
+static bool ggml_amx_initialized = false;
+#endif
+
```

Comment:
Should we move this flag inside the `ggml-amx` implementation?

```c
    ggml_amx_init();

    ...

    if (ggml_amx_is_init()) {
        ...
    }
```

Reply:
ok I will give it a try, not sure if this is feasible.

Reply:
changed in https://github.com/ggerganov/llama.cpp/pull/8998
i moved `ggml_amx_init()` from ggml.c to ggml-amx/mmq.cpp, it will be called for the 1st time launch.